packages/angular_devkit/build_webpack/src/plugins/architect.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { BuilderContext } from '@angular-devkit/architect/src/index2';
+import { strings } from '@angular-devkit/core';
+import * as path from 'path';
+import * as webpack from 'webpack';
+
+/**
+ * This list was taken from a list of events we want to listen to, from the list of hooks in
+ * webpack's typings.
+ */
+const hookSafelist: (keyof webpack.compilation.CompilationHooks)[] = [
+  'seal',
+  'optimizeDependenciesBasic',
+  'optimizeDependencies',
+  'optimizeDependenciesAdvanced',
+  'afterOptimizeDependencies',
+  'optimize',
+  'optimizeModulesBasic',
+  'optimizeModules',
+  'optimizeModulesAdvanced',
+  'afterOptimizeModules',
+  'optimizeChunksBasic',
+  'optimizeChunks',
+  'optimizeChunksAdvanced',
+  'afterOptimizeChunks',
+  'optimizeTree',
+  'afterOptimizeTree',
+  'optimizeChunkModulesBasic',
+  'optimizeChunkModules',
+  'optimizeChunkModulesAdvanced',
+  'afterOptimizeChunkModules',
+  'reviveModules',
+  'optimizeModuleOrder',
+  'advancedOptimizeModuleOrder',
+  'beforeModuleIds',
+  'moduleIds',
+  'optimizeModuleIds',
+  'afterOptimizeModuleIds',
+  'reviveChunks',
+  'optimizeChunkOrder',
+  'beforeChunkIds',
+  'optimizeChunkIds',
+  'afterOptimizeChunkIds',
+  'recordModules',
+  'recordChunks',
+  'beforeHash',
+  'afterHash',
+  'recordHash',
+  'record',
+  'beforeModuleAssets',
+  'shouldGenerateChunkAssets',
+  'beforeChunkAssets',
+  'additionalChunkAssets',
+  'additionalAssets',
+  'optimizeChunkAssets',
+  'afterOptimizeChunkAssets',
+  'optimizeAssets',
+  'afterOptimizeAssets',
+  'afterSeal',
+  'optimizeExtractedChunksBasic',
+  'optimizeExtractedChunks',
+  'optimizeExtractedChunksAdvanced',
+  'afterOptimizeExtractedChunks',
+];
+
+/**
+ * A webpack plugin that reports status and progress to Architect.
+ */
+export class ArchitectPlugin {
```

Comment:
<3

Reply:
Yup. It's used all around, makes code way cleaner.